src/wait.js

```diff
@@ -1,8 +1,79 @@
-import waitForExpect from 'wait-for-expect'
+import {
+  newMutationObserver,
+  getDocument,
+  setImmediate,
+  setTimeout,
+  clearTimeout,
+  runWithRealTimers,
+} from './helpers'
 import {getConfig} from './config'
 
-function wait(callback = () => {}, {timeout = getConfig().asyncUtilTimeout, interval = 50} = {}) {
-  return waitForExpect(callback, timeout, interval)
+function wait(
+  callback = () => {},
+  {
+    container = getDocument(),
+    timeout = getConfig().asyncUtilTimeout,
+    interval = 50,
+    mutationObserverOptions = {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      characterData: true,
+    },
+  } = {},
+) {
+  if (interval < 1) interval = 1
+  const maxTries = Math.ceil(timeout / interval)
+  let tries = 0
+  return new Promise((resolve, reject) => {
+    let lastError, lastTimer
+    const overallTimeoutTimer = setTimeout(onTimeout, timeout)
+
+    const observer = newMutationObserver(checkCallback)
+    runWithRealTimers(() =>
+      observer.observe(container, mutationObserverOptions),
+    )
+
+    function onDone(error, result) {
```

Comment:
Looks like `onDone` is expecting 2 params, but we will never send (and use) them both at the same time.

```js
function onDone(result, cb) {
  clearTimeout(overallTimeoutTimer)
  clearTimeout(lastTimer)
  setImmediate(() => observer.disconnect()) 
  cb(result)  
}
```

...and then: 

```js
onDone(callback(), resolve)
```

```js
onDone(lastError || new Error('Timed out in wait.'), reject)
```

What if we passed the callback action as a parameter, too, so we remove simplify both its public API and its implementation? Not a big deal, just something that crossed my mind.

_disclaimer:_ I'm not using my computer right now so I couldn't give it a go.

Reply:
Sorry, I'm not sure I understand what you're suggesting.

Reply:
If I got it right, `onDone` receives `error` and `result`, and depending on the received param (it's either one or the other, never the two of them) it calls `reject` or `resolve`.

What I was suggesting is that `onDone` received the appropiate payload (either `error` or `result`) and the appropiate callback function (either `reject` or `resolve`).

It's a small refactor, so not a big deal :)

Reply:
Oooohhhh, I understand. Sorry, I was just confused. Yes. I think that's a good idea 👍

Reply:
I think the point is to be consistent with how node callbacks work ("error-first callbacks"). Then `promisify` "just" works. Otherwise the first argument becomes polymorphic which I'd like to avoid as much as possible.